### DISCUSSION TITLE
Proper way to send payload data in event from Parent to Child Machine?

### DISCUSSION THREAD

**Discussion by mmontoya** [statelyai/xstate#1921] [Q&A] [answered]
Hello folks,

I am truly loving working with Xstate, it such a welcomed relief from other state management solutions I have used in the past (*cough* Redux). 

I am attempting to set up a "Practice" machine that depends on an invoked child timer machine. While it has been smooth sailing so far, there is just one detail I am getting hung up on, which I haven't been able to resolve and for which I can't find an answer in the documentation.

One of the possible actions on the timer (available in all states) is to reset its duration. For that, I need to be able to send  the new duration down to the timer as an event payload so that it can assign the new value to its context. While it seems rather straight-forward, the way I have it set up, seems to strip the payload data from the event (even though I can see the event and payload being fired in the visualizer window). I have set up an example that logs the event when the timer receives it and, as you can see, it is devoid of any information but the type property.

Can a kind soul please shed some light on what is the proper way to do this?

Much obliged,

Mike
[CodeSandbox](https://codesandbox.io/s/amazing-dewdney-w0z0i?fontsize=14&hidenavigation=1&theme=dar)

**Comment by Andarist** [comment, marked as answer]
Your practice machine receives `{ type: 'DURATION', value: 5000 }` and based on that it decides to send a `{ type: 'DURATION' }` to a timer machine. Even though both of those events share the very same type they are not the same. This is not forwarding the very same event but rather sending a new event in response to some event. You should just make sure that you copy all required properties onto it - take a look at the adjusted example:
https://codesandbox.io/s/friendly-grass-3bj61?file=/src/index.js

**Reply by mmontoya** [reply]
Ahhhhhhhhh! 
Yes, that makes perfect sense because we are dealing with reference types. :-)
(You know, at some point, I actually entered your amended code into the visualizer, but it kept crashing it, so I figured I was doing something wrong).
Thank you!
